xbmc/pvr/PVRManager.cpp

```diff
@@ -627,15 +627,16 @@ bool CPVRManager::Load(void)
   /* start the add-on update thread */
   if (m_addons)
     m_addons->Start();
+  else
+    return false;
 
-  /* load at least one client */
-  while (IsInitialising() && m_addons && !m_addons->HasConnectedClients())
-    Sleep(50);
```

Comment:
you really don't want to wait for all add-ons to be initialised and connected, but should continue to load when one of them is initialised, and add new data if there are more add-ons to be initialised. with this code, one add-on with a failing connection (e.g. backend offline) will block pvr from being loaded completely.